modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/Marshaller.cs

```diff
@@ -109,6 +108,10 @@ public Marshaller(BinaryConfiguration cfg, ILogger log = null)
             if (typeNames != null)
                 foreach (string typeName in typeNames)
                     AddUserType(new BinaryTypeConfiguration(typeName), typeResolver);
+
+            _registerSameJavaType = _cfg.NameMapper == null || 
```

Comment:
As we agreed on the [dev list](http://apache-ignite-developers.2346864.n4.nabble.com/DISCUSSION-Net-BinaryTypes-transparency-tp50998p51029.html), `_registerSameJavaType` should take effect only within Java service call or `ExecuteJavaTask`.

Reply:
It seems I get you wrong in the dev-list discussion.
Do you suggest to avoid implicit type registration for cache operations?
I think it's questionable from the user's point of view.

Reply:
Yes, I suggest to register Java classes only when Java code is being invoked with `ExecuteJavaTask` or Java service call. I do not think it is necessary for cache operations. Let's get back on the dev list if you'd like to discuss further.

Reply:
OK. I removed changes for Cache API.
We will discuss it later in the follow-ups of this PR.